tally_ho/apps/tally/static/js/barcode_verify.js

```diff
@@ -1,27 +1,60 @@
 function barcodes_match(empty_message, length_message, mismatch_message) {
     var barcode = document.getElementById('id_barcode');
     var barcode_copy = document.getElementById('id_barcode_copy');
+    const barcode_manual_entry = () => document.getElementById('barcode_manual_entry');
+    if (barcode_manual_entry() != null){
+        if (barcode.value == barcode_copy.value) {
+            barcode_copy.parentNode.setAttribute('class', 'has-success');
+            return true;
+        }
+        barcode_copy.parentNode.setAttribute('class', 'has-error');
 
-    if (barcode.value == barcode_copy.value) {
-        barcode_copy.parentNode.setAttribute('class', 'has-success');
-        return true;
-    }
-    barcode_copy.parentNode.setAttribute('class', 'has-error');
-
-    if (barcode.value == "" || barcode_copy.value == "") {
-        barcode.parentNode.setAttribute('class', 'has-error');
-        alert(empty_message);
-    } else if (barcode.value.length < 1) {
-        barcode.parentNode.setAttribute('class', 'has-error');
-        alert(length_message);
-    }
-    else {
-        alert(mismatch_message);
+        if (barcode.value == "" || barcode_copy.value == "") {
+            barcode.parentNode.setAttribute('class', 'has-error');
+            alert(empty_message);
+        } else if (barcode.value.length < 1) {
+            barcode.parentNode.setAttribute('class', 'has-error');
+            alert(length_message);
+        }
+        else {
+            alert(mismatch_message);
+        }
+    }else{
+        return;
     }
 
     return false;
 }
 
+var barcode_number = '';
```

Comment:
seems like not using var is a thing, https://google.github.io/styleguide/jsguide.html#features-use-const-and-let

Reply:
yeah seen that, will use a let instead and move it inside the function it's being used.